qiskit_experiments/randomized_benchmarking/rb_experiment.py

```diff
@@ -123,18 +123,21 @@ def _generate_circuit(self, elements: Iterable[Clifford], lengths: Iterable[int]
         qubits = list(range(self.num_qubits))
         circuits = []
 
-        circ = QuantumCircuit(self.num_qubits)
-        circ.barrier(qubits)
+        circs = [QuantumCircuit(self.num_qubits) for _ in range(len(lengths))]
+        for circ in circs:
+            circ.barrier(qubits)
         circ_op = Clifford(np.eye(2 * self.num_qubits))
 
         for current_length, group_elt in enumerate(elements):
             circ_op = circ_op.compose(group_elt)
-            circ.append(group_elt, qubits)
-            circ.barrier(qubits)
+            group_elt_instruction = group_elt.to_instruction()
+            for circ in circs:
+                circ.append(group_elt_instruction, qubits)
+                circ.barrier(qubits)
             if current_length + 1 in lengths:
```

Comment:
Probably not a lot of overhead, but still if we're optimizing: instead of searching in O(n) using the `in` operator (which is called in each iteration here), you can just compare to the first element and then pop it when it is hit (assuming the list is ordered)